hibernate-core/src/main/java/org/hibernate/dialect/aggregate/OracleAggregateSupport.java

```diff
@@ -68,22 +70,27 @@ public class OracleAggregateSupport extends AggregateSupportImpl {
 
 	private final boolean checkConstraintSupport;
 	private final JsonSupport jsonSupport;
+	private final boolean dateTypesStoreAsString;
 
-	private OracleAggregateSupport(boolean checkConstraintSupport, JsonSupport jsonSupport) {
+	OracleAggregateSupport(boolean checkConstraintSupport, JsonSupport jsonSupport, boolean dateTypesStoreAsString) {
 		this.checkConstraintSupport = checkConstraintSupport;
 		this.jsonSupport = jsonSupport;
+		// this flag tell us if data is serialized/de-serialized as String. As opposed to using OSON
+		// In other words, this flag tells us if the Oracle OSON JDBC extension is used or not.
+		this.dateTypesStoreAsString = dateTypesStoreAsString;
 	}
 
-	public static AggregateSupport valueOf(Dialect dialect) {
+	public static AggregateSupport valueOf(Dialect dialect, boolean useDateStoredAsString) {
```

Comment:
I wonder if the flag is really needed here. It seems to me that we can safely assume `useDateStoredAsString = dialect.getVersion().isBefore( 23 )` i.e. the various SQL fragments you use in case we expect the OSON encoding are compatible with the string encoding.